pkg/feature/manifest_test.go

```diff
@@ -0,0 +1,172 @@
+package feature_test
+
+import (
+	"io/fs"
+	"path/filepath"
+	"testing"
+
+	"github.com/spf13/afero"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ fs.FS = (*AferoFsAdapter)(nil)
+
+type AferoFsAdapter struct {
+	Afs afero.Fs
+}
+
+// Open adapts the Open method to comply with fs.FS interface.
+func (a AferoFsAdapter) Open(name string) (fs.File, error) {
+	return a.Afs.Open(name)
+}
+
+var _ = Describe("Manifest Processing", func() {
+	var (
+		inMemFS AferoFsAdapter
+		path    string
+	)
+
+	BeforeEach(func() {
+		fSys := afero.NewMemMapFs()
+		inMemFS = AferoFsAdapter{Afs: fSys}
+
+	})
+
+	Describe("baseManifest Process", func() {
+		BeforeEach(func() {
+			resourceYaml := `
+apiVersion: v1
+kind: ConfigMap
+metadata:
+ name: my-configmap
+ namespace: fake-ns
+data:
+ key: value
+`
+			path = "path/to/test.yaml"
+			err := afero.WriteFile(inMemFS.Afs, path, []byte(resourceYaml), 0644)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should process the manifest correctly", func() {
+			// given
+			manifest := feature.CreateBaseManifestFrom(inMemFS, path)
+
+			data := feature.Spec{
+				TargetNamespace: "not-used",
+			}
+
+			// when
+			// Simulate adding to and processing from a slice of Manifest interfaces
+			manifests := []feature.Manifest{manifest}
+			objs := processManifests(data, manifests)
+
+			Expect(objs).To(HaveLen(1))
+			Expect(objs[0].GetKind()).To(Equal("ConfigMap"))
+			Expect(objs[0].GetName()).To(Equal("my-configmap"))
+		})
+	})
+
+	Describe("TemplateManifest Process", func() {
+		BeforeEach(func() {
+			resourceYaml := `
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-configmap
+  namespace: {{.TargetNamespace}}
+data:
+  key: Data
+`
+			path = "path/to/template.yaml"
+			err := afero.WriteFile(inMemFS.Afs, path, []byte(resourceYaml), 0644)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should process the template manifest correctly", func() {
```

Comment:
nit, but how this test description is better than `It("should work")` ;) Try to be more accurate with expectations from the test. This description can serve as "API" documentation. 

```suggestion
		It("should substitute target namespace in the templated manifest", func() {
```